### DISCUSSION TITLE
how to save SAM/BAM files with STARsolo with loop

### DISCUSSION THREAD

**Discussion by Polligator** [alexdobin/STAR#1654] [Q&A] [answered]
hi there,  
Thanks for developing this great tool, I have a question here but could not find an answer, hope someone can help.
I tried to loop through a few 10X samples using STARsolo in shell, first, wanted to save both the velocyto files and BAM files, the velocyto files were saved without problems using --soloOutFileNames, however, the BAM files get overwritten while looping through the samples, only the last BAM was saved after the loop under the root folder. tried to use --outFileNamePrefix to provide file names for the separate BAM files, but it errored out.
what would be the correct way to do this?
another question is: what if I just want to save the velocyto file but not the BAM, is there a way to completely skip saving the BAM file?
Thanks

**Comment by alexdobin** [comment, marked as answer]
Hi @joyduck 

if you are looping over samples, it's best to either:
* create new directory for every sample, cd to it, and run STAR from it
OR
* use different ```--outFileNamePrefix``` for each sample
Both of these will save separate files for each run.

If you do not need BAM, you can switch off SAM/BAM output with ```--outSAMtype None```. All other files will be output.

**Reply by Polligator** [reply]
Thanks so much!
